### PR TITLE
Add XML doc comments for Microsoft.Net.Http.Headers

### PR DESCRIPTION
**Pull request by pranavkm**
* Enable missing doc comments to appear as warnings when building locally.
* Nullability changes, minor refactoring

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -100,10 +100,11 @@
 
   <!-- Warnings and errors -->
   <PropertyGroup>
+    <!-- Ensure API docs are available. -->
+    <NoWarn>$(NoWarn.Replace('1591', ''))</NoWarn>
+
     <!-- For local builds, don't make missing XML docs a fatal build error, but still surface so we have visibility into undocumented APIs. -->
     <WarningsNotAsErrors Condition=" '$(ContinuousIntegrationBuild)' != 'true' ">$(WarningsNotAsErrors);CS1591</WarningsNotAsErrors>
-    <!-- For CI builds, ensure API docs are available. -->
-    <NoWarn Condition=" '$(ContinuousIntegrationBuild)' == 'true' ">$(NoWarn.Replace('1591', ''))</NoWarn>
```
Comment (pranavkm) [inline Directory.Build.props:106]: @JunTaoLuo I was missing for the later half of  the conversation here. Having no hints during local development is pretty miserable. With this change, you get warnings (but not errors) for missing doc comments locally, and errors on the CI. 

Does that seem reasonable?

File: src/Http/Headers/src/ContentRangeHeaderValue.cs [hunk 2]
```diff
@@ -74,46 +87,60 @@ public ContentRangeHeaderValue(long from, long to)
...
+
+        /// <summary>
+        /// Gets the end of the range.
+        /// </summary>
+        public long? To { get; private set; }
+
+        /// <summary>
+        /// Gets the total size of the document.
+        /// </summary>
+        [NotNullIfNotNull(nameof(Length))]
```
Comment (pranavkm) [inline src/Http/Headers/src/ContentRangeHeaderValue.cs:122]:
```suggestion
```

File: src/Http/Headers/src/CookieHeaderValue.cs [hunk 3]
```diff
@@ -269,6 +322,7 @@ public override bool Equals(object? obj)
                 && StringSegment.Equals(_value, other._value, StringComparison.OrdinalIgnoreCase);
         }
 
+        /// <inheritdoc />
         public override int GetHashCode()
         {
             return _name.GetHashCode() ^ _value.GetHashCode();
```
Comment (pranavkm) [inline src/Http/Headers/src/CookieHeaderValue.cs:328]: 🐛 these should probably use the same comparer as the one in Equals

**Review by JamesNK — APPROVED** [review]

File: Directory.Build.props (hunk 1 above)
Comment (JunTaoLuo) [reply Directory.Build.props:106]: Seems good.

**Pull request merged**